azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement.go

```diff
@@ -0,0 +1,386 @@
+package memberentitlementmanagement
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/ahmetb/go-linq"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/graph"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/identity"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/licensing"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/memberentitlementmanagement"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/webapi"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/client"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/suppress"
+)
+
+var (
+	groupConfigurationKeys = []string{
+		"origin_id",
+		"origin",
+		"principal_name",
+	}
+)
+
+func ResourceGroupEntitlement() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGroupEntitlementCreate,
+		Read:   resourceGroupEntitlementRead,
+		Delete: resourceGroupEntitlementDelete,
+		Update: resourceGroupEntitlementUpdate,
+		Importer: &schema.ResourceImporter{
+			State: importGroupEntitlement,
+		},
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:         schema.TypeString,
+				Computed:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"principal_name": {
+				Type:          schema.TypeString,
+				Computed:      true,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"origin_id", "origin", "display_name"},
+				AtLeastOneOf:  groupConfigurationKeys,
+				ValidateFunc:  validation.StringIsNotWhiteSpace,
+			},
+			"origin_id": {
+				Type:          schema.TypeString,
+				Computed:      true,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"principal_name"},
+				AtLeastOneOf:  groupConfigurationKeys,
+				ValidateFunc:  validation.StringIsNotWhiteSpace,
+			},
+			"origin": {
+				Type:          schema.TypeString,
+				Computed:      true,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"principal_name"},
+				AtLeastOneOf:  groupConfigurationKeys,
+				ValidateFunc:  validation.StringIsNotWhiteSpace,
+			},
+			"account_license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  licensing.AccountLicenseTypeValues.Express,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(licensing.AccountLicenseTypeValues.Advanced),
+					string(licensing.AccountLicenseTypeValues.EarlyAdopter),
+					string(licensing.AccountLicenseTypeValues.Express),
+					"basic",
```

Comment:
`basic` is the alias of `express`,  it can be replaced by `express`. In web UI page, `basic` is the display name of `express`
![image](https://user-images.githubusercontent.com/57888764/220321005-1e1938f5-1696-468b-8f96-7f629f748f3a.png)